packages/manager/cypress/e2e/core/cloudpulse/alert-notification-channel-list.spec.ts

```diff
@@ -0,0 +1,356 @@
+          );
+        }
+
+        // Alerts list
+        expect(item.alerts.length).to.eq(expected.alerts.length);
+
+        item.alerts.forEach((alert, aIndex) => {
+          const expAlert = expected.alerts[aIndex];
+
+          expect(alert.id).to.eq(expAlert.id);
+          expect(alert.label).to.eq(expAlert.label);
+          expect(alert.type).to.eq(expAlert.type);
+          expect(alert.url).to.eq(expAlert.url);
+        });
+      });
+    });
+  });
+
+  it('searches and validates notification channel details', () => {
+    cy.findByPlaceholderText('Search for Notification Channels').as(
+      'searchInput'
+    );
+
+    cy.get('[data-qa="notification-channels-table"]')
+      .find('tbody')
+      .last()
+      .within(() => {
+        cy.get('tr').should('have.length', 26);
+      });
+
+    cy.get('@searchInput').clear();
+    cy.get('@searchInput').type('Channel-9');
+    cy.get('[data-qa="notification-channels-table"]')
+      .find('tbody')
+      .last()
+      .within(() => {
+        cy.get('tr').should('have.length', 1);
+
+        cy.get('tr').each(($row) => {
+          const expected = notificationChannels[8];
+
+          cy.wrap($row).within(() => {
+            cy.findByText(expected.label).should('be.visible');
+            cy.findByText(String(expected.alerts.length)).should('be.visible');
+            cy.findByText('Email').should('be.visible');
+            cy.get('td').eq(3).should('have.text', expected.created_by);
+            cy.findByText(
+              formatDate(expected.updated, {
+                format: 'MMM dd, yyyy, h:mm a',
+                timezone: 'GMT',
+              })
+            ).should('be.visible');
+            cy.get('td').eq(5).should('have.text', expected.updated_by);
+          });
+        });
+      });
+  });
+
+  it('sorting and validates notification channel details', () => {
+    const sortColumns = [
+      {
+        column: 'Channel Name',
+        ascending: [
+          1, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 2, 20, 21, 22, 23, 24, 25,
+          26, 3, 4, 5, 6, 7, 8, 9,
+        ],
+        descending: [
+          9, 8, 7, 6, 5, 4, 3, 26, 25, 24, 23, 22, 21, 20, 2, 19, 18, 17, 16,
+          15, 14, 13, 12, 11, 10, 1,
+        ],
+      },
+      {
+        column: 'Alerts',
+        ascending: [
+          2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 1, 3, 5, 7, 9, 11, 13,
+          15, 17, 19, 21, 23, 25,
+        ],
+        descending: [
+          1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 2, 4, 6, 8, 10, 12, 14,
+          16, 18, 20, 22, 24, 26,
+        ],
+      },
+      {
+        column: 'Channel Type',
+        ascending: [
+          1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 2, 4, 6, 8, 10, 12, 14,
+          16, 18, 20, 22, 24, 26,
+        ],
+        descending: [
+          2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 1, 3, 5, 7, 9, 11, 13,
+          15, 17, 19, 21, 23, 25,
+        ],
+      },
```

Comment:
@agorthi-akamai , lets see instead of passing hardcoded number, if we can achieve something like this

```suggestion
        column: 'Channel Type',
        ascending: notificationChannels
          .sort((a, b) => a.channel_type.localeCompare(b.channel_type))
          .map((ch) => ch.id),
        descending: notificationChannels
          .sort((a, b) => b.channel_type.localeCompare(a.channel_type))
          .map((ch) => ch.id),
 },
```